v0/destinations/pinterestConversion/transform.js

```diff
@@ -0,0 +1,260 @@
+    throw new CustomError(
+      "It is required at least one of em, hashed_maids or pair of client_ip_address and client_user_agent.",
+      400
+    );
+  }
+  const processedUserPayload = processUserPayload(userPayload);
+  const deducedEventName = deduceEventName(message, Config);
+
+  const finalPayload = {
+    ...processedCommonPayload,
+    event_name: deducedEventName,
+    app_id: appId,
+    advertiser_id: advertiserId,
+    user_data: processedUserPayload
+  };
+  return finalPayload;
+};
+
+/*
+    TODO: 
+
+    Need to check if custom properties can be supported in cloud mode as well. 
+    There is no mention of it, in the documentation. In case, if we can send, 
+    will add that.
+ */
+const trackResponseBuilder = (message, destination) => {
+  const contentArray = [];
+  const contentIds = [];
+  const { properties } = message;
+  let totalQuantity = 0;
+  const mandatoryPayload = identifyPageResponseBuilder(message, destination);
+  // ref: https://s.pinimg.com/ct/docs/conversions_api/dist/v3.html
+  let customPayload = constructPayload(message, CUSTOM_CONFIGS);
+
+  // if product array is present will look for the product level information
+  if (
+    properties.products &&
+    Array.isArray(properties.products) &&
+    properties.products.length > 0
+  ) {
+    const { products } = properties;
+    products.forEach(product => {
+      const prodParams = setIdPriceQuantity(product, message);
+      contentIds.push(prodParams.contentId);
+      contentArray.push(prodParams.content);
+      totalQuantity = product.quantity
+        ? totalQuantity + product.quantity
+        : totalQuantity;
+    });
+
+    if (totalQuantity === 0) {
+      /* 
+      in case any of the products inside product array does not have quantity,
+       will map the quantity of root level 
+      */
+      totalQuantity = properties.quantity;
+    }
+  } else {
+    /* 
+    for the events where product array is not present, root level id, price and
+    quantity are taken into consideration 
+    */
+    const prodParams = setIdPriceQuantity(message.properties, message);
+    contentIds.push(prodParams.contentId);
+    contentArray.push(prodParams.content);
+    totalQuantity = properties.quantity
+      ? totalQuantity + properties.quantity
+      : totalQuantity;
+  }
+  /*
+    if properties.numOfItems is not provided by the user, the total quantity of the products
+    will be sent as num_items
+  */
+  if (!isDefinedAndNotNull(customPayload.num_items)) {
+    customPayload.num_items = parseInt(totalQuantity, 10);
+  }
+  customPayload = {
+    ...customPayload,
+    content_ids: contentIds,
+    contents: contentArray
+  };
+
+  const finalTrackPayload = {
+    ...mandatoryPayload,
+    custom_data: { ...customPayload }
+  };
+  return finalTrackPayload;
+};
+const process = event => {
+  let response;
+  const { message, destination } = event;
+  const messageType = message.type?.toLowerCase();
+  const destConfig = destination.Config;
```

Comment:
Why not object destructuring? And we used destConfig once only.

Reply:
done